src/sign_workflow/sign_artifacts.py

```diff
@@ -23,14 +23,16 @@ class SignArtifacts:
     artifact_type: str
     signature_type: str
     platform: str
+    overwrite: bool
     signer: Signer
 
-    def __init__(self, target: Path, components: List[str], artifact_type: str, signature_type: str, platform: str) -> None:
+    def __init__(self, target: Path, components: List[str], artifact_type: str, signature_type: str, platform: str, overwrite: bool) -> None:
```

Comment:
I think this one doesn't need to take `overwrite`. It will be passed into `sign(... overwrite)` later, it's not a property of a signer I think.